Add env vars to support shl creator and server logging

daniellrgn opened the pull request.

daniellrgn committed:
Add env vars to support shl creator and server logging

ivan-c commented on base/docker-compose.yaml:
```diff
@@ -209,6 +212,7 @@ services:
     image: ghcr.io/uwcirg/shl-ltt-server:${SHL_SERVER_IMAGE_TAG:-latest}
     environment:
       PUBLIC_URL: https://shl-server.${BASE_DOMAIN}
+      VERSION: ${SHL_SERVER_IMAGE_TAG:-"latest"}
```
The docker image tag won't provide much information on development servers, since they generally won't be tagged. You may want to use `git describe` output at build-time to populate this value.

Here's a PR the implements the CI, but you'll also need a way to consume that environment variable in the app code

uwcirg/asbi-screening-app#48

daniellrgn replied:
Thanks! Added that function to the builds here:
shl-creator: https://github.com/uwcirg/shl-ltt/pull/27
shl-server: https://github.com/uwcirg/shl-ltt-server/pull/3
and removed the use of the image tag in this change.

ivan-c commented on base/docker-compose.yaml:
```diff
@@ -209,6 +212,7 @@ services:
     image: ghcr.io/uwcirg/shl-ltt-server:${SHL_SERVER_IMAGE_TAG:-latest}
     environment:
       PUBLIC_URL: https://shl-server.${BASE_DOMAIN}
+      VERSION: ${SHL_SERVER_IMAGE_TAG:-"latest"}
```
This shouldn't need to be quoted, at least for YAML
```suggestion
      VERSION: ${SHL_SERVER_IMAGE_TAG:-latest}
```

ivan-c commented on base/docker-compose.yaml:
```diff
@@ -197,6 +197,9 @@ services:
       VITE_POST_LOGOUT_REDIRECT_URI: https://${BASE_DOMAIN}/users
+      VITE_LOG_URL: https://shl-server.${BASE_DOMAIN}/api/log
+      VITE_SYSTEM_URL: https://shl-creator.${BASE_DOMAIN}
+      VITE_VERSION_STRING: ${SHL_CREATOR_IMAGE_TAG:-"latest"}
```
This shouldn't need to be quoted, at least for YAML

```suggestion
      VITE_VERSION_STRING: ${SHL_CREATOR_IMAGE_TAG:-latest}
```

ivan-c reviewed (approved):
Looks good!

Couple nitpicks, but no blockers

daniellrgn committed:
Remove version vars in favor of build-time git vcs args

shl-creator: https://github.com/uwcirg/shl-ltt/pull/27
shl-server: https://github.com/uwcirg/shl-ltt-server/pull/3
